### DISCUSSION TITLE
Update .d.ts when schema.graphql is changed?

### DISCUSSION THREAD

**Discussion by lubosmato** [0no-co/gql.tada#328] [Q&A] [answered]
Is there a way to configure `gql.tada` to do something like a watch operation on `schema.graphql` file changes and re-generate `.d.ts` file everytime there is a change in `schema.graphql`?

Maybe I am asking for solution to "XY problem" but the goal is to have nice developer experience when developing a full-stack feature on monorepo project. 

E.g. On a monorepo consisting of backend and frontend projects a full-stack developer would modify graphql (backend), this modifies `schema.graphql`, then the dev wants to use new graphql modification in frontend project but at this point has to either restart LSP (TS server) or run `gql.tada generate-output` to have "fresh" `.d.ts` file.

**Comment by JoviDeCroock** [comment, marked as answer]
I just tested this and it doesn't require me to restart the LSP, the LSP watches the `.graphql` file correctly for me and regenerates everything. That being said, the programmatic API of our cli-utils should allow you to create this, the CLI at this time does not implement watch.

The code can be seen here, maybe there's a bug for the LSP but we'd need a reproduction/... https://github.com/0no-co/GraphQLSP/blob/main/packages/graphqlsp/src/graphql/getSchema.ts#L150-L163

**Reply by lubosmato** [reply]
Thank you for a test and quick reply.

Nice to hear that this is the way `gql.tada` suppose to behave. I will re-create the repo and test again.

**Reply by lubosmato** [reply]
After setting up same as in `example-tada` example, `.d.ts` is re-generated on schema changes correctly. Thank you again for your support.
